### PR TITLE
WINTERMUTE: Rewrite absolute prefix workarounds block with loop

### PR DESCRIPTION
**Pull request by tobiatesan**
This one adds a commit on top of #695, it's the result of a few words exchanged with @somaen.

It replaces the large if(){} block which only grew larger in #695 with a loop.

See if it turns out useful.

### CODE REVIEW

File: engines/wintermute/base/file/base_disk_file.cpp [hunk 1]
```diff
@@ -115,11 +115,24 @@ Common::SeekableReadStream *openDiskFile(const Common::String &filename) {
 
 	// Absolute path: TODO: Add specific fallbacks here.
 	if (fixedFilename.contains(':')) {
-		if (fixedFilename.hasPrefix("c:/windows/fonts/")) { // East Side Story refers to "c:\windows\fonts\framd.ttf"
-			fixedFilename = filename.c_str() + 14;
-		} else if (fixedFilename.hasPrefix("c:/carol6/svn/data/")) {	// Carol Reed 6: Black Circle refers to "c:\carol6\svn\data\sprites\system\help.png"
-			fixedFilename = fixedFilename.c_str() + 19;
-		} else {
+		// We try a few workarounds for known broken absolute paths in game scripts...
+		const char* const knownPrefixes[] = {
```
Comment (sev-) [inline engines/wintermute/base/file/base_disk_file.cpp]: Is there a sense to move this to the very top of the file? If so, then we also need to elaborate a bit more in the comment.

Comment (tobiatesan) [inline engines/wintermute/base/file/base_disk_file.cpp]: I'm agnostic on the issue.

Comment (tobiatesan) [inline engines/wintermute/base/file/base_disk_file.cpp]: On a second thought, I think it's slightly more readable right where it is.
I tried moving it and... I liked it slightly less.
If you strongly feel about the issue I don't object either way.

However, I added a // HACK comment to improve discoverability of... the hack.

**Comment by tobiatesan** [comment]
By the way, about that whole block.
It seems to me that the original interpreter simply disregards hardcoded absolute paths and goes on with its business.
@somaen, I was wondering if there is a particular reason why we want to error out at all?
Not that I'm questioning the choice (it is a perfectly legitimate implementation for what is essentially undefined behaviour :) ), just curious.

**Comment by sev-** [comment]
My question got dropped from the conversation due to an outdated commit.

Will it make sense to move that list on the very top of the file, making it easier to expand in the future?

**Comment by tobiatesan** [comment]
Sorry about that @sev-.
So did my answer:  I think it's slightly more readable right where it is.
I tried moving it and... I liked it slightly less.
I am pretty agnostic on the issue, though. If you strongly feel about the issue I don't object either way.

**Comment by sev-** [comment]
No, it was an open question. I'm merging this.

**Pull request merged**